tf5muxserver/mux_server_GetResourceIdentitySchemas.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tf5muxserver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+
+	"github.com/hashicorp/terraform-plugin-mux/internal/logging"
+)
+
+// GetResourceIdentitySchemas merges the schemas returned by the
+// tfprotov5.ResourceIdentitySchema associated with muxServer into a single schema.
+// Everything must be returned from only one server.
+// Schemas must be identical between all servers.
+func (s *muxServer) GetResourceIdentitySchemas(ctx context.Context, req *tfprotov5.GetResourceIdentitySchemasRequest) (*tfprotov5.GetResourceIdentitySchemasResponse, error) {
+	rpc := "GetResourceIdentitySchemas"
+	ctx = logging.InitContext(ctx)
+	ctx = logging.RpcContext(ctx, rpc)
+
+	s.serverDiscoveryMutex.Lock()
+	defer s.serverDiscoveryMutex.Unlock()
+
+	resp := &tfprotov5.GetResourceIdentitySchemasResponse{
+		IdentitySchemas: map[string]*tfprotov5.ResourceIdentitySchema{},
+		Diagnostics:     []*tfprotov5.Diagnostic{},
+	}
+
+	for _, server := range s.servers {
+		ctx := logging.Tfprotov5ProviderServerContext(ctx, server)
+		logging.MuxTrace(ctx, "calling downstream server")
+
+		// TODO: Remove and call server.GetResourceIdentitySchemas below directly once interface becomes required.
+		//nolint:staticcheck // Intentionally verifying interface implementation
+		resourceIdentityServer, ok := server.(tfprotov5.ProviderServerWithResourceIdentity)
+
+		if !ok {
+			resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+				Severity: tfprotov5.DiagnosticSeverityError,
+				Summary:  "GetResourceIdentitySchemas Not Implemented",
+				Detail: "A GetResourceIdentitySchemas call was received by the provider, however the provider does not implement GetResourceIdentitySchemas. " +
+					"Either upgrade the provider to a version that implements GetResourceIdentitySchemas or this is a bug in Terraform that should be reported to the Terraform maintainers.",
+			})
+
+			continue
+		}
+
+		resourceIdentitySchemas, err := resourceIdentityServer.GetResourceIdentitySchemas(ctx, req)
+
+		if err != nil {
+			return resp, fmt.Errorf("error calling GetResourceIdentitySchemas for %T: %w", server, err)
+		}
+
+		resp.Diagnostics = append(resp.Diagnostics, resourceIdentitySchemas.Diagnostics...)
+
+		for resourceIdentityType, schema := range resourceIdentitySchemas.IdentitySchemas {
+			if _, ok := resp.IdentitySchemas[resourceIdentityType]; ok {
+				resp.Diagnostics = append(resp.Diagnostics, resourceIdentityDuplicateError(resourceIdentityType))
+
+				continue
+			}
+
+			resp.IdentitySchemas[resourceIdentityType] = schema
+		}
+	}
+
+	s.serverDiscoveryComplete = true
```

Comment:
> Same comment goes for `tf6muxserver`

```suggestion
```

I'm pretty sure we need to remove this, since the `GetResourceIdentitySchemas` doesn't actually discover resources/data sources. What I believe you would observe, is by calling `GetResourceIdentitySchemas` prior to `GetProviderSchema`, the entire mux server would start returning "resource type not found" because the server discovery would be reported as complete. 

This would be similar to the `GetFunctions` call: https://github.com/hashicorp/terraform-plugin-mux/blob/00c75980e1bc952c6c526a1d8358169cf46c785a/tf6muxserver/mux_server_GetFunctions.go#L54-L55


---------

In reality, I think `GetProviderSchema` will always be called first, but that assumption may not hold up forever 😆